ktor-server/ktor-server-core/jvm/src/io/ktor/features/CORS.kt

```diff
@@ -42,6 +43,11 @@ public class CORS(configuration: Configuration) {
             if (configuration.allowNonSimpleContentTypes) headers else headers.minus(HttpHeaders.ContentType)
         }
 
+    /**
+     * Prefix for permitted headers
+     */
+    public val headerPredicates : List<Predicate<String>> = configuration.headerPredicates
```

Comment:
Depending on Java-only `Predicate` is unnecessary. Can't we replace it with Kotlin function `(String) -> Boolean`? It looks a bit more verbose but it will let us migrate out of JVM in case..